### PR TITLE
Implements placeholder `#if`

### PR DESCRIPTION
**Pull request by ike709**
It's really annoying having to do this every time I want to mess with a real codebase. Tested and working.

### CODE REVIEW

**Comment by ike709** [comment]
image
image

**Comment by ike709** [comment]
Actually DNM, I found an issue while testing against NTstation

**Comment by ike709** [comment]
Ready for review

File: OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessor.cs
Comment (wixoaGit) [inline OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessor.cs:161]: This won't output the warning token if the line is otherwise empty
```suggestion
                        _result.Add(new Token(TokenType.Warning, token.Text, token.SourceFile, token.Line, token.Column, "#if is not implemented"));
```

**Review by wixoaGit — CHANGES_REQUESTED** [review]

File: OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessor.cs
Comment (ike709) [reply OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessor.cs:161]: This PR is blocked until this no longer causes an EOF error.

Comment (wixoaGit) [reply OpenDreamShared/Compiler/DMPreprocessor/DMPreprocessor.cs:161]: As of 8e2523293d6ee8c8f90f967059eb69161a63c17b, this can be merged after the above suggestion.

**Comment by ike709** [comment]
👍🏻

**Pull request merged**
